### PR TITLE
lintcode: 587-Two-Sum-Unique-pairs

### PR DESCRIPTION
**Pull request by wen-wayne-shih**
- lintcode: 587-Two-Sum-Unique-pairs

### CODE REVIEW

File: lintcode/two_pointers/587-Two-Sum-Unique-pairs/solution.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+class Solution:
+    """
+    @param nums: an array of integer
+    @param target: An integer
+    @return: An integer
+    """
+    def twoSum6(self, nums, target):
+        # write your code here
+        if nums is None or len(nums) == 0:
+            return 0
```
Comment (W-Shih) [inline lintcode/two_pointers/587-Two-Sum-Unique-pairs/solution.py:23]:
```suggestion
        if nums is None or len(nums) < 2:
            return 0
```

File: lintcode/two_pointers/587-Two-Sum-Unique-pairs/solution.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        nums.sort()
+
+        count = 0
+        l, r = 0, len(nums) - 1
+        while l < r:
+            two_sum = nums[l] + nums[r]
+            if two_sum == target:
+                count += 1
+                l += 1
+                r -= 1
```
Comment (W-Shih) [inline lintcode/two_pointers/587-Two-Sum-Unique-pairs/solution.py:34]: Make it one line

**Review by W-Shih — CHANGES_REQUESTED** [review]
must be addressed before merging.

File: lintcode/two_pointers/587-Two-Sum-Unique-pairs/solution.py (hunk 1 above)
Comment (wen-wayne-shih) [reply lintcode/two_pointers/587-Two-Sum-Unique-pairs/solution.py]: Resolved as per suggestion

File: lintcode/two_pointers/587-Two-Sum-Unique-pairs/solution.py (hunk 2 above)
Comment (wen-wayne-shih) [reply lintcode/two_pointers/587-Two-Sum-Unique-pairs/solution.py]: Resolved as per suggestion

**Review by W-Shih — CHANGES_REQUESTED** [review]
Looks good. squash commits before merging

**Comment by wen-wayne-shih** [comment]
> Looks good. squash commits before merging

Done

**Review by W-Shih — APPROVED** [review]

**Pull request merged**